pkg/oci/pusher_test.go

```diff
@@ -0,0 +1,16 @@
+package oci
+
+import "testing"
+
+func TestPusher(t *testing.T) {
+	// Create a function
+	// Build the function
+	// Create an HTTP listener on an OS-chosen port ${port}
+	// Create an endpoint which emulates an OCI compliant push-target
+	// Alter the funciton's image tag to be localhost:${port}
+	// Create an OCI pusher instance and .Push
+	// Confirm with the HTTP listener that the expected request was received
+	//  - Only a shallow test is necessary because the unit tests of
+	//    go-containerregistry/pkg/v1/remote ensure it is correct
+	// Note that the test may need to use the explicit username and password
```

Comment:
I agree, and I will either add it as a separate PR or update this PR depending on when the OCI builder (`oci/builder.go`) is available, which is being submitted in a separate PR, and is a bit delayed behind this pusher because it depends on #1728, while this pusher has minimal dependencies on other parts of the scaffolding system.